linode/manager
# 2FA with Security Questions Fixes

## Description 📝

- For existing users with 2FA enabled, we need them to still see the 2FA toggle even if they don't have security questions set
- Pending an API change, a user will not be able to `Reset two-factor authentication` if they don't have security questions setup (shown in screenshots 📷)

## Screenshots 📷

## Bug

![Screen Shot 2022-06-22 at 8 03 27 AM](https://user-images.githubusercontent.com/6440455/175104512-ef9cd9a0-36c1-4cc7-bd74-c37529f51928.jpg)

### Updated

![Screen Shot 2022-06-22 at 11 54 39 AM](https://user-images.githubusercontent.com/6440455/175102751-a5e13eb8-a998-4e6f-bde7-c67fcd7464c0.jpg)
![Screen Shot 2022-06-22 at 11 55 03 AM](https://user-images.githubusercontent.com/6440455/175104039-ab9e71a0-efa9-4fbd-9cba-02db5a379ecc.jpg)
![Screen Shot 2022-06-22 at 11 56 12 AM](https://user-images.githubusercontent.com/6440455/175102775-50928caf-332c-4954-849b-4859f8c2c246.jpg)


## How to test 🧪

- Testing 2FA and Security Questions and their various states
  - If an existing user has 2FA enabled but no security questions, they should be able to see the toggle and disable 2fa
  - If an existing user has 2FA enabled but no security questions and they try to click `Reset two-factor authentication`, the API will (not deployed yet) error (shown in screenshots)
  - If the user does not have 2FA setup and has no security questions, they should see a notice saying to set security questions
